### PR TITLE
docs: #34499 Alternatives to docker, root filesystem permissions

### PR DESCRIPTION
**Pull request by enspritz**
…s in docker.rst

Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by enspritz** [comment]
Motivating issue #34499

This change has two parts:

A) A gentle, gradual step towards the notion of running Envoy using container technologies other than Docker. Specifically mentioned by name is Podman.

B) Express indication that Envoy must be run with its root filesystem in read-write mode. From a security perspective, read-only is much more desirable. So whether this newly-documented read-write requirement is ground truth, I don't know. But it is conservative, is probably established practice, and hopefully won't result in misleading anybody such that they "hold Envoy wrong" or cause damage. What I hope it will do is to enrage some kind soul to the point of motivating them to document exactly how the image can be run in read-only mode ;) My guess is that `/tmp` will need to be a writable `tmpfs` volume mount.

File: docs/root/start/docker.rst [hunk 1]
```diff
@@ -3,6 +3,9 @@
 Using the Envoy Docker Image
 ============================
 
+Envoy OCI images are built using Docker and have been extensively battle-tested in large scale 
+deployments running with Docker. Use of other container technologies such as Podman, while known to
+function correctly, is not expressly supported.
```
Comment (phlax) [inline docs/root/start/docker.rst:8]: do we know it functions correctly? i havent tested this myself and have seen a couple of related issues

File: docs/root/start/docker.rst [hunk 2]
```diff
@@ -50,13 +53,29 @@ Build the Docker image using:
...
+or in Podman (unsupported) with:
+
+.. code-block:: console
+
+   $ podman run -d --name envoy -p 9901:9901 -p 10000:10000 envoy:v1
+
+Root filesystem permissions for running Envoy in containers
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The Envoy container image must be run with the container's root filesystem mounted read-write.
```
Comment (phlax) [inline docs/root/start/docker.rst:71]: i dont think this is correct - i didnt try it, but i dont see any reason why it cant be mounted read-only - unless logging, caching, or some other use of the fs is configured

File: docs/root/start/docker.rst [hunk 3]
```diff
@@ -3,6 +3,9 @@
 Using the Envoy Docker Image
 ============================
 
+Envoy OCI images are built using Docker and have been extensively battle-tested in large scale
```
Comment (phlax) [inline docs/root/start/docker.rst:6]: wondering if we should put this in a `.. note::`

**Review by phlax — COMMENTED** [review]
thanks for working on this @enspritz - couple of questions ...

File: docs/root/start/docker.rst (hunk 2 above)
Comment (enspritz) [reply docs/root/start/docker.rst:71]: Admittedly, it's a conservative take. I don't currently have the latitude to comprehensively test RO myself. And I'm not a person that can assert on behalf of the Envoy project that RO is supported. Unless someone has better info ;)

File: docs/root/start/docker.rst (hunk 1 above)
Comment (enspritz) [reply docs/root/start/docker.rst:8]: Thanks for the info. How about language along the lines of 

"The Envoy project does not expressly support the use of container technologies other than Docker, such as Podman, even though such use might appear to function correctly in your use case."

File: docs/root/start/docker.rst (hunk 2 above)
Comment (phlax) [reply docs/root/start/docker.rst:71]: i think its wrong - for example testing with:

```console
$ docker run -p 10000:10000 -v $PWD/configs/:/configs --read-only envoyproxy/envoy:distroless-v1.30-latest -c /configs/envoy-demo.yaml
```

works as expected

File: docs/root/start/docker.rst (hunk 1 above)
Comment (phlax) [reply docs/root/start/docker.rst:8]: > Use of other container technologies such as Podman may work but have not been extensively testsed and are not expressly supported.

File: docs/root/start/docker.rst (hunk 2 above)
Comment (enspritz) [reply docs/root/start/docker.rst:71]: This is exactly what I want -- Acknowledgement, blessing, instructions, and official support from the Envoy project on running Envoy containers read-only. Lemme update this PR.

**Comment by tyxia** [comment]
/wait 

actualy @enspritz  Please fix the format CI issue first

**Comment by enspritz** [comment]
@tyxia Please check it

**Comment by tyxia** [comment]
/wait-any

Thanks for fixing the CI failure, defer to @phlax for the review. Thanks!

File: docs/root/start/docker.rst [hunk 4]
```diff
@@ -3,7 +3,12 @@
 Using the Envoy Docker Image
 ============================
 
-The following examples use the :ref:`official Envoy Docker image <start_install_docker>`.
+.. note::
+  Envoy OCI images are built using Docker and have been extensively battle-tested in large scale
```
Comment (phlax) [inline docs/root/start/docker.rst:7]:
```suggestion
  Envoy OCI images are built using Docker and have been extensively tested in large scale
```

File: docs/root/start/docker.rst [hunk 5]
```diff
@@ -50,16 +55,30 @@ Build the Docker image using:
...
+
+.. code-block:: console
+
+   $ podman run -d --name envoy -p 9901:9901 -p 10000:10000 envoy:v1
+
+Root filesystem permissions for running Envoy in containers
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The Envoy container image can be run with the container's root filesystem mounted read-only.
+With Docker and Podman, this means using the ``--read-only`` option of the ``run`` command.
```
Comment (phlax) [inline docs/root/start/docker.rst:74]:
```suggestion
For example, using Docker and Podman, you can use the ``--read-only`` option of the ``run`` command.
```

it doesnt necessarily mean using `run` - this should work equally with `up` or in a situation where you dont call the container directly (ie k8s/compose/etc)

Comment (phlax) [inline docs/root/start/docker.rst:74]:
```suggestion

With Docker and Podman, this means using the ``--read-only`` option of the ``run`` command.
```

rst wont render a new line (/paragraph) without an empty line in between

File: docs/root/start/docker.rst [hunk 6]
```diff
@@ -50,16 +55,30 @@ Build the Docker image using:
...
+With Kubernetes, this means setting ``podSpec.containers.securityContext.readOnlyFilesystem`` to ``true``.
+With Nomad, this means setting ``readonly_rootfs = true`` in the task's ``config`` block when using the ``docker`` or ``podman`` driver.
```
Comment (phlax) [inline docs/root/start/docker.rst:76]:
```suggestion

With Kubernetes, this means setting ``podSpec.containers.securityContext.readOnlyFilesystem`` to ``true``.

With Nomad, this means setting ``readonly_rootfs = true`` in the task's ``config`` block when using the ``docker`` or ``podman`` driver.
```

same with these lines

**Review by phlax — COMMENTED** [review]
thanks for iterating @enspritz

a couple of nits and this should be good to land

**Comment by enspritz** [comment]
All suggestions committed, pls check

**Review by phlax — APPROVED** [review]
lgtm, thanks @enspritz

**Pull request merged**